### PR TITLE
first pi60rgb_v2 commit

### PR DESCRIPTION
**Pull request by ziptyze**
Addition of 1upkeyboards pi60rgb v2 keyboard

## Description

This is the first commit of the 1upkeyboards pi60rgb v2 keyboard; a hotswap 60% ANSI keyboard with per-key RGB and underglow RGB. The original version of the pi60rgb did not feature underglow. This is a fully redesigned keyboard with new keyswitch matrix and lighting.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": false,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (waffle87) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:26]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "rgb_matrix": true,
```

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    },
+    "matrix_pins": {
+        "cols": ["GP25", "GP1", "GP15", "GP4", "GP5", "GP14", "GP6", "GP7", "GP8", "GP13", "GP12", "GP9", "GP11", "GP10"],
+        "rows": ["GP29", "GP27", "GP26", "GP17", "GP16"]
+    },
+    "rgb_matrix": {
```
Comment (waffle87) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:32]:
```suggestion
    "rgb_matrix": {
        "max_brightness": 125,
```

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/config.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define RGB_MATRIX_LED_COUNT 104
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 125
```
Comment (waffle87) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/config.h:20]:
```suggestion
```

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                               KC_RALT, KC_RGUI, KC_APP,  KC_RCTL
+    ),
+
+    [1] = LAYOUT_60_ansi(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                         KC_TRNS,                               KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
```
Comment (waffle87) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/keymaps/default/keymap.c:35]:
```suggestion
    )
```

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                         KC_TRNS,                               KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [9] = LAYOUT_60_ansi(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                         KC_TRNS,                               KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (waffle87) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/keymaps/via/keymap.c:98]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+            {"x": 190, "y": 57, "flags": 2},
+
+            {"matrix": [4, 12], "x": 196, "y": 57, "flags": 1},
+
+            {"x": 202, "y": 57, "flags": 2},
+            {"x": 209, "y": 57, "flags": 2},
+
+            {"matrix": [4, 13], "x": 215, "y": 57, "flags": 1},
+
+            {"x": 220, "y": 57, "flags": 2}
```
Comment (drashna) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:229]:
```suggestion
            {"x": 4, "y": 5, "flags": 2},
            {"matrix": [0, 0], "x": 8, "y": 7, "flags": 1},
            {"matrix": [0, 1], "x": 23, "y": 7, "flags": 4},
            {"x": 30, "y": 7, "flags": 2},
            {"matrix": [0, 2], "x": 37, "y": 7, "flags": 4},
            {"x": 45, "y": 7, "flags": 2},
            {"matrix": [0, 3], "x": 52, "y": 7, "flags": 4},
            {"x": 60, "y": 7, "flags": 2},
            {"matrix": [0, 4], "x": 67, "y": 7, "flags": 4},
            {"x": 75, "y": 7, "flags": 2},
            {"matrix": [0, 5], "x": 82, "y": 7, "flags": 4},
            {"x": 90, "y": 7, "flags": 2},
            {"matrix": [0, 6], "x": 97, "y": 7, "flags": 4},
            {"x": 105, "y": 7, "flags": 2},
            {"matrix": [0, 7], "x": 112, "y": 7, "flags": 4},
            {"x": 120, "y": 7, "flags": 2},
            {"matrix": [0, 8], "x": 127, "y": 7, "flags": 4},
            {"x": 135, "y": 7, "flags": 2},
            {"matrix": [0, 9], "x": 142, "y": 7, "flags": 4},
            {"x": 149, "y": 7, "flags": 2},
            {"matrix": [0, 10], "x": 157, "y": 7, "flags": 4},
            {"x": 164, "y": 7, "flags": 2},
            {"matrix": [0, 11], "x": 172, "y": 7, "flags": 4},
            {"x": 179, "y": 7, "flags": 2},
            {"matrix": [0, 12], "x": 187, "y": 7, "flags": 4},
            {"x": 198, "y": 7, "flags": 2},
            {"matrix": [0, 13], "x": 209, "y": 7, "flags": 1},
            {"x": 220, "y": 7, "flags": 2},
            {"x": 220, "y": 19, "flags": 2},
            {"matrix": [1, 13], "x": 213, "y": 19, "flags": 1},
            {"matrix": [1, 12], "x": 194, "y": 19, "flags": 4},
            {"matrix": [1, 11], "x": 179, "y": 19, "flags": 4},
            {"matrix": [1, 10], "x": 164, "y": 19, "flags": 4},
            {"matrix": [1, 9], "x": 149, "y": 19, "flags": 4},
            {"matrix": [1, 8], "x": 134, "y": 19, "flags": 4},
            {"matrix": [1, 7], "x": 119, "y": 19, "flags": 4},
            {"matrix": [1, 6], "x": 105, "y": 19, "flags": 4},
            {"matrix": [1, 5], "x": 90, "y": 19, "flags": 4},
            {"matrix": [1, 4], "x": 75, "y": 19, "flags": 4},
            {"matrix": [1, 3], "x": 60, "y": 19, "flags": 4},
            {"matrix": [1, 2], "x": 45, "y": 19, "flags": 4},
            {"matrix": [1, 1], "x": 30, "y": 19, "flags": 4},
            {"matrix": [1, 0], "x": 11, "y": 19, "flags": 1},
            {"x": 4, "y": 19, "flags": 2},
            {"x": 4, "y": 32, "flags": 2},
            {"matrix": [2, 0], "x": 13, "y": 32, "flags": 1},
            {"matrix": [2, 1], "x": 34, "y": 32, "flags": 4},
            {"matrix": [2, 2], "x": 49, "y": 32, "flags": 4},
            {"matrix": [2, 3], "x": 64, "y": 32, "flags": 4},
            {"matrix": [2, 4], "x": 79, "y": 32, "flags": 4},
            {"matrix": [2, 5], "x": 94, "y": 32, "flags": 4},
            {"matrix": [2, 6], "x": 108, "y": 32, "flags": 4},
            {"matrix": [2, 7], "x": 123, "y": 32, "flags": 4},
            {"matrix": [2, 8], "x": 138, "y": 32, "flags": 4},
            {"matrix": [2, 9], "x": 153, "y": 32, "flags": 4},
            {"matrix": [2, 10], "x": 168, "y": 32, "flags": 4},
            {"matrix": [2, 11], "x": 183, "y": 32, "flags": 4},
            {"matrix": [2, 13], "x": 207, "y": 32, "flags": 1},
            {"x": 220, "y": 32, "flags": 2},
            {"x": 220, "y": 45, "flags": 2},
            {"matrix": [3, 13], "x": 203, "y": 45, "flags": 1},
            {"matrix": [3, 10], "x": 175, "y": 45, "flags": 4},
            {"matrix": [3, 9], "x": 161, "y": 45, "flags": 4},
            {"matrix": [3, 8], "x": 146, "y": 45, "flags": 4},
            {"matrix": [3, 7], "x": 135, "y": 45, "flags": 4},
            {"matrix": [3, 6], "x": 116, "y": 45, "flags": 4},
            {"matrix": [3, 5], "x": 101, "y": 45, "flags": 4},
            {"matrix": [3, 4], "x": 86, "y": 45, "flags": 4},
            {"matrix": [3, 3], "x": 71, "y": 45, "flags": 4},
            {"matrix": [3, 2], "x": 56, "y": 45, "flags": 4},
            {"matrix": [3, 1], "x": 41, "y": 45, "flags": 4},
            {"matrix": [3, 0], "x": 17, "y": 45, "flags": 1},
            {"x": 4, "y": 45, "flags": 2},
            {"x": 4, "y": 57, "flags": 2},
            {"matrix": [4, 0], "x": 9, "y": 57, "flags": 1},
            {"x": 15, "y": 57, "flags": 2},
            {"x": 23, "y": 57, "flags": 2},
            {"matrix": [4, 1], "x": 28, "y": 57, "flags": 1},
            {"x": 34, "y": 57, "flags": 2},
            {"x": 41, "y": 57, "flags": 2},
            {"matrix": [4, 2], "x": 47, "y": 57, "flags": 1},
            {"x": 53, "y": 57, "flags": 2},
            {"x": 67, "y": 57, "flags": 2},
            {"x": 71, "y": 57, "flags": 1},
            {"x": 81, "y": 57, "flags": 2},
            {"x": 95, "y": 57, "flags": 2},
            {"matrix": [4, 5], "x": 103, "y": 57, "flags": 1},
            {"x": 110, "y": 57, "flags": 2},
            {"x": 125, "y": 57, "flags": 2},
            {"x": 134, "y": 57, "flags": 1},
            {"x": 139, "y": 57, "flags": 2},
            {"x": 153, "y": 57, "flags": 2},
            {"matrix": [4, 9], "x": 159, "y": 57, "flags": 1},
            {"x": 165, "y": 57, "flags": 2},
            {"x": 172, "y": 57, "flags": 2},
            {"matrix": [4, 10], "x": 177, "y": 57, "flags": 1},
            {"x": 183, "y": 57, "flags": 2},
            {"x": 190, "y": 57, "flags": 2},
            {"matrix": [4, 12], "x": 196, "y": 57, "flags": 1},
            {"x": 202, "y": 57, "flags": 2},
            {"x": 209, "y": 57, "flags": 2},
            {"matrix": [4, 13], "x": 215, "y": 57, "flags": 1},
            {"x": 220, "y": 57, "flags": 2}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+            {"x": 202, "y": 57, "flags": 2},
+            {"x": 209, "y": 57, "flags": 2},
+            {"matrix": [4, 13], "x": 215, "y": 57, "flags": 1},
+            {"x": 220, "y": 57, "flags": 2}
+        ]
+    },
+    "ws2812": {
+        "driver": "vendor",
+        "pin": "GP28"
+    },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:182]:
```suggestion
    },
    "community_layouts": ["60_ansi"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/config.h [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+#pragma once
+
+#define RGB_MATRIX_LED_COUNT 104
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+
+
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/config.h:24]: tidy
```suggestion
```

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+            "solid_reactive_multicross": true,
+            "solid_reactive_multinexus": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_nexus": true,
+            "solid_reactive_simple": true,
+            "solid_reactive_wide": true,
+            "solid_splash": true,
+            "splash": true,
+            "typing_heatmap": true
+    },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:71]: tidy
```suggestion
        },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by ziptyze** [comment]
The QMK CI build only passes if I remove the community layout configuration from the info.json

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json [hunk 10]
```diff
@@ -0,0 +1,250 @@
...
+            {"x": 202, "y": 57, "flags": 2},
+            {"x": 209, "y": 57, "flags": 2},
+            {"matrix": [4, 13], "x": 215, "y": 57, "flags": 1},
+            {"x": 220, "y": 57, "flags": 2}
+        ]
+    },
+    "ws2812": {
+        "driver": "vendor",
+        "pin": "GP28"
+    },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:182]: CI BuiId failures that are attributed to community layout/userspace keymaps would not be dismissed. What's imperative is the keyboard, configuration, and keymap(s) specific to this PR are functional/working - which they are 😉
```suggestion
    },
    "community_layouts" : ["60_ansi"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json (hunk 10 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:182]: When I put this line back in, then the build fails with this error:

🔴 1upkeyboards/1up_pi60_rgb_v2:brandonschlack-ansi

Comment (dunk2k) [reply keyboards/1upkeyboards/1up_pi60_rgb_v2/info.json:182]: `brandonschlack-ansi` is a community layout keymap; any issues with it, thankfully, are not your responsibility to fix. Please configure/define this keyboard as it's intended 🙂

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md:1]: Could an Imgur hosted image of this PCB be added?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+# pi60_rgb
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md:1]:
```suggestion
# 1up pi60 rgb v2
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md
Comment (ziptyze) [reply keyboards/1upkeyboards/1up_pi60_rgb_v2/readme.md:1]: added

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Per discord: https://discord.com/channels/440868230475677696/867530265433473086/1173654755399184514

1up_ should be removed from the 1up_pi60rgb_v2 name

File: keyboards/1upkeyboards/pi60_rgb_v2/config.h [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define RGB_MATRIX_LED_COUNT 104
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
```
Comment (fauxpark) [inline keyboards/1upkeyboards/pi60_rgb_v2/config.h:22]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/pi60_rgb_v2/config.h (hunk 12 above)
Comment (dunk2k) [reply keyboards/1upkeyboards/pi60_rgb_v2/config.h:22]: @fauxpark 
still need to keep `RGB_MATRIX_LED_COUNT`? 😳

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/pi60_rgb_v2/info.json [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "rgb_matrix": true,
+    },
+    "matrix_pins": {
+        "cols": ["GP25", "GP1", "GP15", "GP4", "GP5", "GP14", "GP6", "GP7", "GP8", "GP13", "GP12", "GP9", "GP11", "GP10"],
+        "rows": ["GP29", "GP27", "GP26", "GP17", "GP16"]
+    },
+    "rgb_matrix": {
```
[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/1upkeyboards/pi60_rgb_v2/info.json:26]: No longer necessary as of https://github.com/qmk/qmk_firmware/pull/21110

File: keyboards/1upkeyboards/pi60_rgb_v2/info.json [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+        "max_brightness": 125,
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60_rgb_v2/info.json:27]:
```suggestion
        "max_brightness": 125,
        "sleep": true,
```

File: keyboards/1upkeyboards/pi60_rgb_v2/info.json [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "layer_count": 10
+    },
+    "features": {
+        "bootmagic": true,
+        "extrakey": true,
+        "mousekey": true,
+        "rgb_matrix": true,
+    },
```
Comment (waffle87) [inline keyboards/1upkeyboards/pi60_rgb_v2/info.json:21]:
```suggestion
    },
    "qmk": {
        "locking": {
            "enabled": true,
            "resync": true
        }
    },
```

File: keyboards/1upkeyboards/pi60_rgb_v2/config.h
Comment (waffle87) [inline keyboards/1upkeyboards/pi60_rgb_v2/config.h:1]: This file should be removed with the above suggestions.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/pi60_rgb_v2/info.json (hunk 15 above)
Comment (dunk2k) [reply keyboards/1upkeyboards/pi60_rgb_v2/info.json:21]: bump

File: keyboards/1upkeyboards/pi60_rgb_v2/config.h
Comment (dunk2k) [reply keyboards/1upkeyboards/pi60_rgb_v2/config.h:1]: bump

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
There are suggestions that have not been addressed.

**Comment by ziptyze** [comment]
> There are suggestions that have not been addressed.

I have removed the file

**Review by fauxpark — APPROVED** [review]

**Pull request merged**